src/qutip_cupy/dense_functions.py

```diff
@@ -0,0 +1,29 @@
+# Contains functions for dense_cupy, this are the same functions that
+# are defined ouside the dense.pyx file
+
+import cupy as cp
+
+from .dense import CuPyDense
+
+
+def tidyup_dense(matrix, tol, inplace=True):
+    return matrix
+
+
+def reshape_cupydense(cp_arr, n_rows_out, n_cols_out):
+
+    return CuPyDense(cp_arr, (n_rows_out, n_cols_out))
```

Comment:
I think here `cp_arr` will be a `CuPyDense` already, so this function should be `CuPyDense._raw_cupy_constructor(cupy.reshape(cp_arr._cp, (n_rows_out, n_cols_out))`?

Reply:
yes, good catch